cmd/tailscale/cli/up.go

```diff
@@ -320,8 +321,13 @@ func prefsFromUpArgs(upArgs upArgsT, warnf logger.Logf, st *ipnstate.Status, goo
 		}
 	}
 
-	if len(upArgs.hostname) > 256 {
-		return nil, fmt.Errorf("hostname too long: %d bytes (max 256)", len(upArgs.hostname))
+	// RFC 1035 says the length of a DNS name is restricted to 255 octets or less.
+	if len(upArgs.hostname) > 255 {
```

Comment:
I might suggest moving the length checks to `dnsname` so that it can use the existing constants there.